### PR TITLE
HWBridge RF transceiver extension

### PR DESCRIPTION
**Pull request by zombieCraig**
Primarily designed and tested using [rfcat](https://bitbucket.org/atlas0fd00m/rfcat) and the [YARD stick one](http://greatscottgadgets.com/yardstickone/) although should work on any device that rfcat supports.  Requires that you run the rfcat_msfrelay.  Currently there is [Mercurial pull request](https://bitbucket.org/atlas0fd00m/rfcat/pull-requests/16/rfcat-server-for-metasploit-hardware/diff#comment-None) for this.  Assuming that PR lands there then the relay will come with the standard rfcat package.  Else we can always include it in Metasploit directly.

This PR adds a complete RF transceiver extension documentation is temporary stored [here](http://opengarages.org/hwbridge/v2/) until this PR lands.  This changes the API version to API 0.0.2.  Many of the commands of rfcat are available interactive through the HWBridge shell.  All those commands plus a few more are available through the Metasploit post module interface.  All rfcat exploit scripts should now be fairly easy to port to Metasploit and two are included in this PR.

## Modules

**transmitter**

Is a basic RF transmitter

**rfpwnon**

RF Bruteforcer for AM/OOK.  Demonstrated against static garage door openers.

## Verification

List the steps needed to make sure this thing works

- [x] Start rfcat_msfrelay
- [x] **Verify** you see: RfCat MSFRelay running.
- [x] Start `msfconsole` (In another terminal window)
- [x] `use auxiliary/client/hwbridge/connect`
- [x] At auxiliary(connect) > `run`
- [x] **Verify** You see the NOTICE: leaving the matrix message
- [x] `sessions`
- [x] **Verify** You see a hwbridge session of type rftransceiver
- [x] `sessions -i 1`
- [x] `status`
- [x] **Verify** The device name reports YARDSTICKONE

### CODE REVIEW

**Comment by pbarry-r7** [comment]
I have a YARD Stick One on order, but will go ahead and grab this (if anyone else with a YARD Stick One in their possession wants to verify, by all means feel free to do so!  :)  ).  Looks like some msftidy grumblings are causing Travis to fail at the moment.

**Comment by zombieCraig** [comment]
@pbarry-r7 I forgot to run msftidy.  Doing that now to clean things up.

**Comment by pbarry-r7** [comment]
I did verify the module works with the YARD Stick One I picked up:

First terminal window (following insertion of YARD Stick One device and using the new rfcat_msfrelay script):
```
$ dmesg|tail
[13226.400069] usb 1-2: SerialNumber: 0000
[13336.227130] ohci-pci 0000:00:06.0: bad entry 35b77181
[13893.674139] usb 1-2: USB disconnect, device number 4
[14333.675161] usb 1-2: new full-speed USB device number 5 using ohci-pci
[14333.927130] usb 1-2: config 1 descriptor has 1 excess byte, ignoring
[14333.944267] usb 1-2: New USB device found, idVendor=1d50, idProduct=605b
[14333.944269] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[14333.944271] usb 1-2: Product: YARD Stick One
[14333.944272] usb 1-2: Manufacturer: Great Scott Gadgets
[14333.944273] usb 1-2: SerialNumber: 0000
$ sudo ./rfcat_msfrelay --noauth
RfCat MSFRelay running.
127.0.0.1 - - [16/Feb/2017 20:04:57] "GET /status HTTP/1.1" 200 -
127.0.0.1 - - [16/Feb/2017 20:05:08] "GET /status HTTP/1.1" 200 -
```

Second terminal window:
```
$ ./msfconsole -q
msf > use auxiliary/client/hwbridge/connect
msf auxiliary(connect) > run

[*] Attempting to connect to 127.0.0.1...
[*] Hardware bridge interface session 1 opened (127.0.0.1 -> 127.0.0.1) at 2017-02-16 20:04:57 -0600
[+] HWBridge session established
[*] HW Specialty: {"rftransceiver"=>true}  Capabilities: {"cc11xx"=>true}
[!] NOTICE:  You are about to leave the matrix.  All actions performed on this hardware bridge
[!]          could have real world consequences.  Use this module in a controlled testing
[!]          environment and with equipment you are authorized to perform testing on.
[*] Auxiliary module execution completed
msf auxiliary(connect) > sessions

Active sessions
===============

  Id  Type                   Information    Connection
  --  ----                   -----------    ----------
  1   hwbridge cmd/hardware  rftransceiver  127.0.0.1 -> 127.0.0.1 (127.0.0.1)

msf auxiliary(connect) > sessions -i 1
[*] Starting interaction with 1...

hwbridge > status
[*] Operational: Yes
[*] Device: YARDSTICKONE
[*] FW Version: 450
[*] HW Version: 0348
```

👍

File: lib/rex/post/hwbridge/ui/console/command_dispatcher/rftransceiver.rb [hunk 1]
```diff
@@ -108,8 +108,8 @@ def cmd_freq(*args)
     arg = {}
     opts = Rex::Parser::Arguments.new(
       '-h' => [ false, 'Help Banner' ],
-      '-f' => [ true, 'frequency to set, example: 433000000'],
-      '-m' => [ true, 'Mhz, default is 24']
```
Comment (pbarry-r7) [inline lib/rex/post/hwbridge/ui/console/command_dispatcher/rftransceiver.rb]: I removed the 'default is 24' here, as I couldn't see where 24 is being used as the default value, but I wanted to double-check that I'm not missing something.

**Comment by todb-r7** [comment]
GTG in the morning. @pbarry-r7. :)

**Comment by pbarry-r7** [comment]
Push rebased branch to reconcile conflicts just now...

**Comment by pbarry-r7** [comment]
# Release Notes

This patch allows the Hardware Bridge to interact with RF transceiver devices that are supported by rfcat.  Additionally, two post modules (transmitter and rfpwnon) ported over from RfCatHelpers scripts are included with this patch.

**Pull request merged**